<HTML><HEAD>
<TITLE>bwa.1</TITLE>

<style type="text/css">
a:link {
	text-decoration: none;
	color: #0092e8;
}
a:visited {
	text-decoration: none;
	color: #0092e8;
}
a:hover {
	text-decoration: underline;
	color: #0092e8;
}
body, td, th {
	font: 12px consolas, andale mono, courier, mono;
}
body {
	color: #000;
	background: #fff;
	margin: 0;
	padding: 0;
}
table {
	border: solid 0px #ccc;
}
td {
	vertical-align: top;
	padding: 0.2em;
}
th {
	font-weight: bold;
	text-align: left;
	padding: 0.2em;
}
#tbl table {
	border: solid 1px #ccc;
}
#tbl td {
	border: solid 1px #ccc;
	padding: 0.3em;
}
#wrap {
  width: 740px;
  text-align: left;
  margin: 0 auto;
}
hr {
	margin: 1em 0;
	color: #C7C7C7;
	background: #C7C7C7;
	border-color: #C7C7C7;
	border-style: none;
	height: 1px;
}
h1, h2, h3, h4, h5, h6 {
	font-family: "Trebuchet MS", arial, sans-serif;
	font-weight: bold;
}
p {
    text-align: justify;
}
</style>
</HEAD>
<BODY>
<div id="wrap"><A NAME=top></A>
<!-- .TH bwa 1 "2 September 2009" "bwa-0.5.1" "Bioinformatics tools" -->
<CENTER>
<H1><HR><I>Manual Reference Pages &nbsp;-&nbsp;</I><NOBR>bwa (1)</NOBR><HR></H1>
</CENTER>
<!-- .SH NAME -->
<A name=0>

     <H3>NAME</H3>

</A>
<BLOCKQUOTE>
<!-- .PP -->
<P>
bwa - Burrows-Wheeler Alignment Tool
<!-- .SH SYNOPSIS -->
</BLOCKQUOTE>
<A name=contents></A><H3>CONTENTS</H3></A>
<BLOCKQUOTE>
<A HREF=#1>Synopsis</A><BR>
<A HREF=#2>Description</A><BR>
<A HREF=#3>Commands And Options</A><BR>
<A HREF=#4>Sam Alignment Format</A><BR>
<A HREF=#5>Notes On Short-read Alignment</A><BR>
&nbsp; &nbsp; &nbsp;&nbsp; &nbsp; &nbsp;<A HREF=#6>Alignment Accuracy</A><BR>
&nbsp; &nbsp; &nbsp;&nbsp; &nbsp; &nbsp;<A HREF=#7>Memory Requirement</A><BR>
&nbsp; &nbsp; &nbsp;&nbsp; &nbsp; &nbsp;<A HREF=#8>Speed</A><BR>
<A HREF=#9>Notes On Long-read Alignment</A><BR>
<A HREF=#10>See Also</A><BR>
<A HREF=#11>Author</A><BR>
<A HREF=#12>License And Citation</A><BR>
<A HREF=#13>History</A><BR>
</BLOCKQUOTE>
<A name=14>

     <H3>SYNOPSIS</H3>

</A>
<BLOCKQUOTE>
<!-- .PP -->
<P>
bwa index -a bwtsw database.fasta
<!-- .PP -->
<P>
bwa aln database.fasta short_read.fastq &gt; aln_sa.sai
<!-- .PP -->
<P>
bwa samse database.fasta aln_sa.sai short_read.fastq &gt; aln.sam
<!-- .PP -->
<P>
bwa sampe database.fasta aln_sa1.sai aln_sa2.sai read1.fq read2.fq &gt; aln.sam
<!-- .PP -->
<P>
bwa dbwtsw database.fasta long_read.fastq &gt; aln.sam
<P>
<!-- .SH DESCRIPTION -->
</BLOCKQUOTE>
<A name=2>

     <H3>DESCRIPTION</H3>

</A>
<BLOCKQUOTE>
<!-- .PP -->
<P>
BWA is a fast light-weighted tool that aligns relatively short sequences
(queries) to a sequence database (targe), such as the human reference
genome. It implements two different algorithms, both based on
Burrows-Wheeler Transform (BWT). The first algorithm is designed for
short queries up to ~200bp with low error rate (&lt;3%). It does gapped
global alignment w.r.t. queries, supports paired-end reads, and is one
of the fastest short read alignment algorithms to date while also
visiting suboptimal hits. The second algorithm, dBWT-SW, is designed for
long reads with more errors. It performs heuristic Smith-Waterman-like
alignment to find high-scoring local hits (and thus chimera). On
low-error short queries, dBWT-SW is slower and less accurate than the
first algorithm, but on long queries, it is better.
<!-- .PP -->
<P>
For both algorithms, the database file in the FASTA format must be
first indexed with the
<!-- .B `index' -->
<B>&#145;index&#146;</B> command, which typically takes a few hours. The first algorithm is
implemented via the
<!-- .B `aln' -->
<B>&#145;aln&#146;</B> command, which finds the suffix array (SA) coordinates of good hits of
each individual read, and the
<!-- .B `samse/sampe' -->
<B>&#145;samse/sampe&#146;</B> command, which converts SA coordinates to chromosomal coordinate and
pairs reads (for &#145;sampe&#146;). The second algorithm is invoked by the
<!-- .B `dbtwsw' -->
<B>&#145;dbtwsw&#146;</B> command. It works for single-end reads only.
<P>
<!-- .SH COMMANDS AND OPTIONS -->
</BLOCKQUOTE>
<A name=3>

     <H3>COMMANDS AND OPTIONS</H3>

</A>
<BLOCKQUOTE>
<!-- .TP -->
<TABLE cellpadding=3>
<TR valign=top><TD width=6% nowrap>
<!-- .B index -->
<B>index</B> </TD><TD valign=bottom>
bwa index [-p prefix] [-a algoType] [-c] &lt;in.db.fasta&gt;
<P>
Index database sequences in the FASTA format.
<P>
<!-- .B OPTIONS: -->
<B>OPTIONS:</B> <!-- .RS -->
<!-- .TP 10 -->
<TABLE width=100% cellpadding=3><!-- tsb: <B>OPTIONS:</B>  -->
<TR></TR><TR></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -c -->
<B>-c</B> </TD><TD valign=bottom>
Build color-space index. The input fast should be in nucleotide space.
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -p STR -->
<B>-p STR</B> </TD><TD valign=bottom>
Prefix of the output database [same as db filename]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -a STR -->
<B>-a STR</B> </TD><TD valign=bottom>
Algorithm for constructing BWT index. Available options are:
<!-- .RS -->
<!-- .TP -->
<TABLE width=100% cellpadding=3><!-- tsb: Algorithm for constructing BWT index. Available options are:
 -->
<TR></TR><TR></TR>
<TR valign=top><TD width=6% nowrap>
<!-- .B is -->
<B>is</B> </TD><TD valign=bottom>
IS linear-time algorithm for constructing suffix array. It requires
5.37N memory where N is the size of the database. IS is moderately fast,
but does not work with database larger than 2GB. IS is the default
algorithm due to its simplicity. The current codes for IS algorithm are
reimplemented by Yuta Mori.
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=6% nowrap>
<!-- .B bwtsw -->
<B>bwtsw</B> </TD><TD valign=bottom>
Algorithm implemented in BWT-SW. This method works with the whole human
genome, but it does not work with database smaller than 10MB and it is
usually slower than IS.
<!-- .RE -->
</TD></TR>
<TR></TR></TABLE><!-- .RE -->
</TD></TR>
<TR></TR></TABLE><P>
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=6% nowrap>
<!-- .B aln -->
<B>aln</B> </TD><TD valign=bottom>
bwa aln [-n maxDiff] [-o maxGapO] [-e maxGapE] [-d nDelTail] [-i
nIndelEnd] [-k maxSeedDiff] [-l seedLen] [-t nThrds] [-cRN] [-M misMsc]
[-O gapOsc] [-E gapEsc] &lt;in.db.fasta&gt; &lt;in.query.fq&gt; &gt; &lt;out.sai&gt;
<P>
Find the SA coordinates of the input reads. Maximum
<!-- .I maxSeedDiff -->
<I>maxSeedDiff</I> differences are allowed in the first
<!-- .I seedLen -->
<I>seedLen</I> subsequence and maximum
<!-- .I maxDiff -->
<I>maxDiff</I> differences are allowed in the whole sequence.
<P>
<!-- .B OPTIONS: -->
<B>OPTIONS:</B> <!-- .RS -->
<!-- .TP 10 -->
<TABLE width=100% cellpadding=3><!-- tsb: <B>OPTIONS:</B>  -->
<TR></TR><TR></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -n NUM -->
<B>-n NUM</B> </TD><TD valign=bottom>
Maximum edit distance if the value is INT, or the fraction of missing
alignments given 2% uniform base error rate if FLOAT. In the latter
case, the maximum edit distance is automatically chosen for different
read lengths. [0.04]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -o INT -->
<B>-o INT</B> </TD><TD valign=bottom>
Maximum number of gap opens [1]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -e INT -->
<B>-e INT</B> </TD><TD valign=bottom>
Maximum number of gap extensions, -1 for k-difference mode (disallowing
long gaps) [-1]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -d INT -->
<B>-d INT</B> </TD><TD valign=bottom>
Disallow a long deletion within INT bp towards the 3&#146;-end [16]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -i INT -->
<B>-i INT</B> </TD><TD valign=bottom>
Disallow an indel within INT bp towards the ends [5]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -l INT -->
<B>-l INT</B> </TD><TD valign=bottom>
Take the first INT subsequence as seed. If INT is larger than the query
sequence, seeding will be disabled. For long reads, this option is
typically ranged from 25 to 35 for &#145;-k 2&#146;. [inf]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -k INT -->
<B>-k INT</B> </TD><TD valign=bottom>
Maximum edit distance in the seed [2]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -t INT -->
<B>-t INT</B> </TD><TD valign=bottom>
Number of threads (multi-threading mode) [1]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -M INT -->
<B>-M INT</B> </TD><TD valign=bottom>
Mismatch penalty. BWA will not search for suboptimal hits with a score
lower than (bestScore-misMsc). [3]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -O INT -->
<B>-O INT</B> </TD><TD valign=bottom>
Gap open penalty [11]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -E INT -->
<B>-E INT</B> </TD><TD valign=bottom>
Gap extension penalty [4]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -R INT -->
<B>-R INT</B> </TD><TD valign=bottom>
Proceed with suboptimal alignments if there are no more than INT equally
best hits. This option only affects paired-end mapping. Increasing this
threshold helps to improve the pairing accuracy at the cost of speed,
especially for short reads (~32bp).
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -c -->
<B>-c</B> </TD><TD valign=bottom>
Reverse query but not complement it, which is required for alignment in
the color space.
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -N -->
<B>-N</B> </TD><TD valign=bottom>
Disable iterative search. All hits with no more than
<!-- .I maxDiff -->
<I>maxDiff</I> differences will be found. This mode is much slower than the default.
<!-- .RE -->
</TD></TR>
<TR></TR></TABLE><P>
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B samse -->
<B>samse</B> </TD><TD valign=bottom>
bwa samse [-n maxOcc] &lt;in.db.fasta&gt; &lt;in.sai&gt; &lt;in.fq&gt; &gt; &lt;out.sam&gt;
<P>
Generate alignments in the SAM format given single-end reads. Repetitive
hits will be randomly chosen. When
<!-- .B -n -->
<B>-n</B> is applied, multiple positions will be printed in a concise format. The
&#145;&gt;&#146; line in the output shows the read name, the number of printed hits
and the number of stored hits; the following lines give the chromosomal
coordinates and the edit distance.
<P>
<!-- .B OPTIONS: -->
<B>OPTIONS:</B> <!-- .RS -->
<!-- .TP 10 -->
<TABLE width=100% cellpadding=3><!-- tsb: <B>OPTIONS:</B>  -->
<TR></TR><TR></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -n INT -->
<B>-n INT</B> </TD><TD valign=bottom>
Output up to INT top hits. Value -1 to disable outputting multiple
hits. [-1]
<!-- .RE -->
</TD></TR>
<TR></TR></TABLE><P>
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B sampe -->
<B>sampe</B> </TD><TD valign=bottom>
bwa sampe [-a maxInsSize] [-o maxOcc] &lt;in.db.fasta&gt; &lt;in1.sai&gt; &lt;in2.sai&gt;
&lt;in1.fq&gt; &lt;in2.fq&gt; &gt; &lt;out.sam&gt;
<P>
Generate alignments in the SAM format given paired-end reads. Repetitive
read pairs will be placed randomly.
<P>
<!-- .B OPTIONS: -->
<B>OPTIONS:</B> <!-- .RS -->
<!-- .TP 8 -->
<TABLE width=100% cellpadding=3><!-- tsb: <B>OPTIONS:</B>  -->
<TR></TR><TR></TR>
<TR valign=top><TD width=8% nowrap>
<!-- .B -a INT -->
<B>-a INT</B> </TD><TD valign=bottom>
Maximum insert size for a read pair to be considered being mapped
properly. Since 0.4.5, this option is only used when there are not
enough good alignment to infer the distribution of insert sizes. [500]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=8% nowrap>
<!-- .B -o INT -->
<B>-o INT</B> </TD><TD valign=bottom>
Maximum occurrences of a read for pairing. A read with more occurrneces
will be treated as a single-end read. Reducing this parameter helps
faster pairing. [100000]
<!-- .RE -->
</TD></TR>
<TR></TR></TABLE><P>
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=8% nowrap>
<!-- .B dbwtsw -->
<B>dbwtsw</B> </TD><TD valign=bottom>
bwa dbwtsw [-a matchScore] [-b mmPen] [-q gapOpenPen] [-r gapExtPen] [-t
nThreads] [-w bandWidth] [-T thres] [-s hspIntv] [-z zBest] [-N
nHspRev] [-c thresCoef] &lt;in.db.fasta&gt; &lt;in.fq&gt;
<P>
Align query sequences in the &lt;in.fq&gt; file.
<P>
<!-- .B OPTIONS: -->
<B>OPTIONS:</B> <!-- .RS -->
<!-- .TP 10 -->
<TABLE width=100% cellpadding=3><!-- tsb: <B>OPTIONS:</B>  -->
<TR></TR><TR></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -a INT -->
<B>-a INT</B> </TD><TD valign=bottom>
Score of a match [1]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -b INT -->
<B>-b INT</B> </TD><TD valign=bottom>
Mismatch penalty [3]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -q INT -->
<B>-q INT</B> </TD><TD valign=bottom>
Gap open penalty [5]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -r INT -->
<B>-r INT</B> </TD><TD valign=bottom>
Gap extension penalty. The penalty for a contiguous gap of size k is
q+k*r. [2]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -t INT -->
<B>-t INT</B> </TD><TD valign=bottom>
Number of threads in the multi-threading mode [1]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -w INT -->
<B>-w INT</B> </TD><TD valign=bottom>
Band width in the banded alignment [33]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -T INT -->
<B>-T INT</B> </TD><TD valign=bottom>
Minimum score threshold divided by a [37]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -c FLOAT -->
<B>-c FLOAT</B> </TD><TD valign=bottom>
Coefficient for threshold adjustment according to query length. Given an
l-long query, the threshold for a hit to be retained is
a*max{T,c*log(l)}. [5.5]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -z INT -->
<B>-z INT</B> </TD><TD valign=bottom>
Z-best heuristics. Higher -z increases accuracy at the cost of speed. [1]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -s INT -->
<B>-s INT</B> </TD><TD valign=bottom>
Maximum SA interval size for initiating a seed. Higher -s increases
accuracy at the cost of speed. [3]
<!-- .TP -->
</TD></TR>
<TR valign=top><TD width=10% nowrap>
<!-- .B -N INT -->
<B>-N INT</B> </TD><TD valign=bottom>
Minimum number of seeds supporting the resultant alignment to skip
reverse alignment. [5]
<!-- .RE -->
</TD></TR>
<TR></TR></TABLE><P>
<!-- .SH SAM ALIGNMENT FORMAT -->
</TD></TR>
<TR></TR></TABLE></BLOCKQUOTE>
<A name=4>

     <H3>SAM ALIGNMENT FORMAT</H3>

</A>
<BLOCKQUOTE>
<!-- .PP -->
<P>
The output of the
<!-- .B `aln' -->
<B>&#145;aln&#146;</B> command is binary and designed for BWA use only. BWA outputs the final
alignment in the SAM (Sequence Alignment/Map) format. Each line consists
of:
<P>
<!-- .TS -->
<P><BLOCKQUOTE><div id="tbl"><TABLE cellspacing=1>
<!-- center box; -->
<!-- cb | cb | cb -->
<!-- n | l | l . -->
<TR valign=top>
<TD align=center><B>Col</B></TD><TD align=center><B>Field</B></TD><TD align=center><B>Description</B></TD></TR>
<TR></TR><TR></TR>
<TR valign=top>
<TD align=right>1</TD><TD>QNAME</TD><TD>Query (pair) NAME</TD></TR>
<TR valign=top>
<TD align=right>2</TD><TD>FLAG</TD><TD>bitwise FLAG</TD></TR>
<TR valign=top>
<TD align=right>3</TD><TD>RNAME</TD><TD>Reference sequence NAME</TD></TR>
<TR valign=top>
<TD align=right>4</TD><TD>POS</TD><TD>1-based leftmost POSition/coordinate of clipped sequence</TD></TR>
<TR valign=top>
<TD align=right>5</TD><TD>MAPQ</TD><TD>MAPping Quality (Phred-scaled)</TD></TR>
<TR valign=top>
<TD align=right>6</TD><TD>CIAGR</TD><TD>extended CIGAR string</TD></TR>
<TR valign=top>
<TD align=right>7</TD><TD>MRNM</TD><TD>Mate Reference sequence NaMe (&#145;=&#146; if same as RNAME)</TD></TR>
<TR valign=top>
<TD align=right>8</TD><TD>MPOS</TD><TD>1-based Mate POSistion</TD></TR>
<TR valign=top>
<TD align=right>9</TD><TD>ISIZE</TD><TD>Inferred insert SIZE</TD></TR>
<TR valign=top>
<TD align=right>10</TD><TD>SEQ</TD><TD>query SEQuence on the same strand as the reference</TD></TR>
<TR valign=top>
<TD align=right>11</TD><TD>QUAL</TD><TD>query QUALity (ASCII-33 gives the Phred base quality)</TD></TR>
<TR valign=top>
<TD align=right>12</TD><TD>OPT</TD><TD>variable OPTional fields in the format TAG:VTYPE:VALUE</TD></TR>
</TABLE></div></BLOCKQUOTE>
<P>
<!-- .PP -->
<P>
Each bit in the FLAG field is defined as:
<P>
<!-- .TS -->
<P><BLOCKQUOTE><div id="tbl"><TABLE cellspacing=1>
<!-- center box; -->
<!-- cb | cb | cb -->
<!-- c | l | l . -->
<TR valign=top>
<TD align=center><B>Chr</B></TD><TD align=center><B>Flag</B></TD><TD align=center><B>Description</B></TD></TR>
<TR></TR><TR></TR>
<TR valign=top>
<TD align=center>p</TD><TD>0x0001</TD><TD>the read is paired in sequencing</TD></TR>
<TR valign=top>
<TD align=center>P</TD><TD>0x0002</TD><TD>the read is mapped in a proper pair</TD></TR>
<TR valign=top>
<TD align=center>u</TD><TD>0x0004</TD><TD>the query sequence itself is unmapped</TD></TR>
<TR valign=top>
<TD align=center>U</TD><TD>0x0008</TD><TD>the mate is unmapped</TD></TR>
<TR valign=top>
<TD align=center>r</TD><TD>0x0010</TD><TD>strand of the query (1 for reverse)</TD></TR>
<TR valign=top>
<TD align=center>R</TD><TD>0x0020</TD><TD>strand of the mate</TD></TR>
<TR valign=top>
<TD align=center>1</TD><TD>0x0040</TD><TD>the read is the first read in a pair</TD></TR>
<TR valign=top>
<TD align=center>2</TD><TD>0x0080</TD><TD>the read is the second read in a pair</TD></TR>
<TR valign=top>
<TD align=center>s</TD><TD>0x0100</TD><TD>the alignment is not primary</TD></TR>
<TR valign=top>
<TD align=center>f</TD><TD>0x0200</TD><TD>QC failure</TD></TR>
<TR valign=top>
<TD align=center>d</TD><TD>0x0400</TD><TD>optical or PCR duplicate</TD></TR>
</TABLE></div></BLOCKQUOTE>
<P>
<!-- .PP -->
<P>
The Please check &lt;<A HREF="http://samtools.sourceforge.net">http://samtools.sourceforge.net</A>&gt; for the format
specification and the tools for post-processing the alignment.
<P>
BWA generates the following optional fields. Tags starting with &#145;X&#146; are
specific to BWA.
<P>
<!-- .TS -->
<P><BLOCKQUOTE><div id="tbl"><TABLE cellspacing=1>
<!-- center box; -->
<!-- cb | cb -->
<!-- cB | l . -->
<TR valign=top>
<TD align=center><B>Tag</B></TD><TD align=center><B>Meaning</B></TD></TR>
<TR></TR><TR></TR>
<TR valign=top>
<TD align=center><B>NM</B></TD><TD>Edit distance</TD></TR>
<TR valign=top>
<TD align=center><B>MD</B></TD><TD>Mismatching positions/bases</TD></TR>
<TR valign=top>
<TD align=center><B>AS</B></TD><TD>Alignment score</TD></TR>
<TR></TR><TR></TR>
<TR valign=top>
<TD align=center><B>X0</B></TD><TD>Number of best hits</TD></TR>
<TR valign=top>
<TD align=center><B>X1</B></TD><TD>Number of suboptimal hits found by BWA</TD></TR>
<TR valign=top>
<TD align=center><B>XN</B></TD><TD>Number of ambiguous bases in the referenece</TD></TR>
<TR valign=top>
<TD align=center><B>XM</B></TD><TD>Number of mismatches in the alignment</TD></TR>
<TR valign=top>
<TD align=center><B>XO</B></TD><TD>Number of gap opens</TD></TR>
<TR valign=top>
<TD align=center><B>XG</B></TD><TD>Number of gap extentions</TD></TR>
<TR valign=top>
<TD align=center><B>XT</B></TD><TD>Type: Unique/Repeat/N/Mate-sw</TD></TR>
<TR></TR><TR></TR>
<TR valign=top>
<TD align=center><B>XS</B></TD><TD>Suboptimal alignment score</TD></TR>
<TR valign=top>
<TD align=center><B>XF</B></TD><TD>Support from forward/reverse alignment</TD></TR>
<TR valign=top>
<TD align=center><B>XE</B></TD><TD>Number of supporting seeds</TD></TR>
</TABLE></div></BLOCKQUOTE>
<P>
<!-- .PP -->
<P>
Note that XO and XG are generated by BWT search while the CIGAR string
by Smith-Waterman alignment. These two tags may be inconsistent with the
CIGAR string. This is not a bug.
<P>
<!-- .SH NOTES ON SHORT-READ ALIGNMENT -->
</BLOCKQUOTE>
<A name=5>

     <H3>NOTES ON SHORT-READ ALIGNMENT</H3>

</A>
<BLOCKQUOTE>
<!-- .SS Alignment Accuracy -->
</BLOCKQUOTE>
<A name=6>

    <H4>&nbsp; &nbsp; Alignment Accuracy</H4>
</A>
<BLOCKQUOTE>
<!-- .PP -->
<P>
When seeding is disabled, BWA guarantees to find an alignment
containing maximum
<!-- .I maxDiff -->
<I>maxDiff</I> differences including
<!-- .I maxGapO -->
<I>maxGapO</I> gap opens which do not occur within
<!-- .I nIndelEnd -->
<I>nIndelEnd</I> bp towards either end of the query. Longer gaps may be found if
<!-- .I maxGapE -->
<I>maxGapE</I> is positive, but it is not guaranteed to find all hits. When seeding is
enabled, BWA further requires that the first
<!-- .I seedLen -->
<I>seedLen</I> subsequence contains no more than
<!-- .I maxSeedDiff -->
<I>maxSeedDiff</I> differences.
<!-- .PP -->
<P>
When gapped alignment is disabled, BWA is expected to generate the same
alignment as Eland, the Illumina alignment program. However, as BWA
change &#145;N&#146; in the database sequence to random nucleotides, hits to these
random sequences will also be counted. As a consequence, BWA may mark a
unique hit as a repeat, if the random sequences happen to be identical
to the sequences which should be unqiue in the database. This random
behaviour will be avoided in future releases.
<!-- .PP -->
<P>
By default, if the best hit is no so repetitive (controlled by -R), BWA
also finds all hits contains one more mismatch; otherwise, BWA finds all
equally best hits only. Base quality is NOT considered in evaluating
hits. In pairing, BWA searches, among the found hits under the
constraint of the
<!-- .I maxOcc -->
<I>maxOcc</I> option, for pairs within
<!-- .I maxInsSize -->
<I>maxInsSize</I> and with proper orientation.
<P>
<!-- .SS Memory Requirement -->
</BLOCKQUOTE>
<A name=7>

    <H4>&nbsp; &nbsp; Memory Requirement</H4>
</A>
<BLOCKQUOTE>
<!-- .PP -->
<P>
With bwtsw algorithm, 2.5GB memory is required for indexing the complete
human genome sequences. For short reads, the
<!-- .B `aln' -->
<B>&#145;aln&#146;</B> command uses ~2.3GB memory and the
<!-- .B `sampe' -->
<B>&#145;sampe&#146;</B> command uses ~3.5GB.
<P>
<!-- .SS Speed -->
</BLOCKQUOTE>
<A name=8>

    <H4>&nbsp; &nbsp; Speed</H4>
</A>
<BLOCKQUOTE>
<!-- .PP -->
<P>
Indexing the human genome sequences takes 3 hours with bwtsw
algorithm. Indexing smaller genomes with IS or divsufsort algorithms is
several times faster, but requires more memory.
<!-- .PP -->
<P>
Speed of alignment is largely determined by the error rate of the query
sequences (r). Firstly, BWA runs much faster for near perfect hits than
for hits with many differences, and it stops searching for a hit with
l+2 differences if a l-difference hit is found. This means BWA will be
very slow if r is high because in this case BWA has to visit hits with
many differences and looking for these hits is expensive. Secondly, the
alignment algorithm behind makes the speed sensitive to [k log(N)/m],
where k is the maximum allowed differences, N the size of database and m
the length of a query. In practice, we choose k w.r.t. r and therefore r
is the leading factor. I would not recommend to use BWA on data with
r&gt;0.02.
<!-- .PP -->
<P>
Pairing is slower for shorter reads. This is mainly because shorter
reads have more spurious hits and converting SA coordinates to
chromosomal coordinates are very costly.
<!-- .PP -->
<P>
In a practical experiment, BWA is able to map 2 million 32bp reads to a
bacterial genome in several minutes, map the same amount of reads to
human X chromosome in 8-15 minutes and to the human genome in 15-25
minutes. This result implies that the speed of BWA is insensitive to the
size of database and therefore BWA is more efficient when the database
is sufficiently large. On smaller genomes, hash based algorithms are
usually much faster.
<P>
<!-- .SH NOTES ON LONG-READ ALIGNMENT -->
</BLOCKQUOTE>
<A name=9>

     <H3>NOTES ON LONG-READ ALIGNMENT</H3>

</A>
<BLOCKQUOTE>
<!-- .PP -->
<P>
Command
<!-- .B `dbwtsw' -->
<B>&#145;dbwtsw&#146;</B> is designed for long-read alignment. The algorithm behind, dBWT-SW, is
similar to BWT-SW, but does not guarantee to find all local hits due to
the heuristic acceleration. It tends to be faster and more accurate if
the resultant alignment is supported by more seeds, and therefore
dBWT-SW usually performs better on long queries than on short ones.
<P>
On 350-1000bp reads, dBWT-SW is several to tens of times faster than the
existing programs. Its accuracy is comparable to SSAHA2, more accurate
than BLAT. Like BLAT, dBWT-SW also finds chimera which may pose a
challenge to SSAHA2. On 10-100kbp queries where chimera detection is
important, dBWT-SW is over 10X faster than BLAT while being more
sensitive.
<P>
DBWT-SW can also be used to align ~100bp reads, but it is slower than
the short-read algorithm. Its sensitivity and accuracy is lower than
SSAHA2 especially when the sequencing error rate is above 2%. This is
the trade-off of the 30X speed up in comparison to SSAHA2&#146;s -454 mode.
<P>
<!-- .SH SEE ALSO -->
</BLOCKQUOTE>
<A name=10>

     <H3>SEE ALSO</H3>

</A>
<BLOCKQUOTE>
BWA website &lt;<A HREF="http://bio-bwa.sourceforge.net">http://bio-bwa.sourceforge.net</A>&gt;, Samtools website
&lt;<A HREF="http://samtools.sourceforge.net">http://samtools.sourceforge.net</A>&gt;
<P>
<!-- .SH AUTHOR -->
</BLOCKQUOTE>
<A name=11>

     <H3>AUTHOR</H3>

</A>
<BLOCKQUOTE>
Heng Li at the Sanger Institute wrote the key source codes and
integrated the following codes for BWT construction: bwtsw
&lt;<A HREF="http://i.cs.hku.hk/~ckwong3/bwtsw">http://i.cs.hku.hk/~ckwong3/bwtsw</A>/&gt;, implemented by Chi-Kwong Wong at
the University of Hong Kong and IS
&lt;<A HREF="http://yuta.256.googlepages.com/sais">http://yuta.256.googlepages.com/sais</A>&gt; originally proposed by Nong Ge
&lt;<A HREF="http://www.cs.sysu.edu.cn/nong">http://www.cs.sysu.edu.cn/nong</A>/&gt; at the Sun Yat-Sen University and
implemented by Yuta Mori.
<P>
<!-- .SH LICENSE AND CITATION -->
</BLOCKQUOTE>
<A name=12>

     <H3>LICENSE AND CITATION</H3>

</A>
<BLOCKQUOTE>
<!-- .PP -->
<P>
The full BWA package is distributed under GPLv3 as it uses source codes
from BWT-SW which is covered by GPL. Sorting, hash table, BWT and IS
libraries are distributed under the MIT license.
<!-- .PP -->
<P>
If you use the short-read alignment component, please cite the following
paper:
<!-- .PP -->
<P>
Li H. and Durbin R. (2009) Fast and accurate short read alignment with
Burrows-Wheeler transform. Bioinformatics, 25, 1754-60.
<P>
<!-- .SH HISTORY -->
</BLOCKQUOTE>
<A name=13>

     <H3>HISTORY</H3>

</A>
<BLOCKQUOTE>
BWA is largely influenced by BWT-SW. It uses source codes from BWT-SW
and mimics its binary file formats; dBWT-SW resembles BWT-SW in several
ways. The initial idea about BWT-based alignment also came from the
group who developed BWT-SW. At the same time, BWA is different enough
from BWT-SW. The short-read alignment algorithm bears no similarity to
Smith-Waterman algorithm any more. While dBWT-SW learns from BWT-SW, it
introduces heuristics that can hardly be applied to the original
algorithm. In all, BWA does not guarantee to find all local hits as what
BWT-SW is designed to do, but it is much faster than BWT-SW on both
short and long query sequences.
<P>
I started to write the first piece of codes on 24 May 2008 and got the
initial stable version on 02 June 2008. During this period, I was
acquainted that Professor Tak-Wah Lam, the first author of BWT-SW paper,
was collaborating with Beijing Genomics Institute on SOAP2, the successor
to SOAP (Short Oligonucleotide Analysis Package). SOAP2 has come out in
November 2008. According to the SourceForge download page, the third
BWT-based short read aligner, bowtie, was first released in August
2008. At the time of writing this manual, at least three more BWT-based
short-read aligners are being implemented.
<P>
The dBWT-SW algorithm is a new component of BWA. It was conceived in
November 2008 and implemented ten months later.
</BLOCKQUOTE>
<P><HR>
<TABLE width=100%><TR> <TD width=33%><I>bwa-0.5.1</I></TD> <TD width=33% align=center>bwa (1)</TD> <TD align=right width=33%><I>2 September 2009</I></TD> </TR></TABLE></div></BODY></HTML>
